sui/contract/sources/constant_product.move

```diff
@@ -89,154 +144,201 @@ module aux::constant_product {
         )
     }
 
-    /// Swap coin_x for coin_y (ETH-USDC would be +1 ETH & -2000 USDC)
-    public fun swap_exact_x_for_y<X, Y, W>(
-        self: &mut Pool<X, Y, W>, 
+    entry fun swap_exact_y_for_x_<X, Y>(
+        self: &mut Pool<X, Y>,
+        coin_y: Coin<Y>,
+        min_base_amount: u64,
+        ctx: &mut TxContext
+    ) {
+        transfer::transfer(
+            swap_exact_y_for_x(self, coin_y, min_base_amount, ctx),
+            tx_context::sender(ctx)
+        )
+    }
+
+    entry fun add_liquidity_<X, Y>(
+        self: &mut Pool<X, Y>,
+        coin_x: Coin<X>,
+        coin_y: Coin<Y>,
+        ctx: &mut TxContext
+    ) {
+        transfer::transfer(
+            add_liquidity(self, coin_x, coin_y, ctx),
+            tx_context::sender(ctx)
+        );
+    }
+
+    entry fun remove_liquidity_<X, Y>(
+        self: &mut Pool<X, Y>,
+        lp: Coin<LP<X, Y>>,
+        ctx: &mut TxContext
+    ) {
+        let (coin_x, coin_y) = remove_liquidity(self, lp, ctx);
+        let sender = tx_context::sender(ctx);
+
+        transfer::transfer(coin_x, sender);
+        transfer::transfer(coin_y, sender);
+    }
+
+    /********************/
+    /* Public functions */
+    /********************/
+
+    public fun create_pool<X, Y>(
```

Comment:
Why is this separate from the entry function? Seems like they have the same signature.

Reply:
you're right, `create_pool` can be combined